packages/lit-node-client-nodejs/src/lib/helpers/validate-bls-session-sig.ts

```diff
@@ -32,6 +34,34 @@ export const blsSessionSigVerify = (
   );
   const signatureBytes = Buffer.from(sigJson.ProofOfPossession, `hex`);
 
+  /** Check time or now */
+  const checkTime = new Date();
+
+  if (!authSigSiweMessage.expirationTime || !authSigSiweMessage.notBefore) {
+    throw new Error(
+      'Invalid SIWE message. Missing expirationTime or notBefore.'
+    );
+  }
+
+  // check timestamp of SIWE
+  const expirationDate = new Date(authSigSiweMessage.expirationTime);
+  if (checkTime.getTime() >= expirationDate.getTime()) {
+    throw new SiweError(
+      SiweErrorType.EXPIRED_MESSAGE,
+      `${checkTime.toISOString()} < ${expirationDate.toISOString()}`,
+      `${checkTime.toISOString()} >= ${expirationDate.toISOString()}`
+    );
+  }
+
+  const notBefore = new Date(authSigSiweMessage.notBefore);
+  if (checkTime.getTime() < notBefore.getTime()) {
+    throw new SiweError(
+      SiweErrorType.NOT_YET_VALID_MESSAGE,
+      `${checkTime.toISOString()} >= ${notBefore.toISOString()}`,
+      `${checkTime.toISOString()} < ${notBefore.toISOString()}`
+    );
+  }
```

Comment:
Should we be checking on expiration in this scope? as this type of exp check is relevant for all sig types and not just BLS?

Reply:
the other sig type (ecdsa) is already checked by the SIWE verify function so i think we're good

Reply:
This makes sense. Will approve ❤️